lib/improver/tests/nowcasting/optical_flow/test_AdvectField.py

```diff
@@ -396,6 +405,47 @@ def test_lead_time(self):
         self.assertEqual(len(lead_time), 1)
         self.assertEqual(lead_time[0], self.timestep.total_seconds())
 
+    def test_units_and_datatypes_for_time_coordinates(self):
+        """Test that the output cube contains the desired units and datatypes
+        for the time, forecast_reference_time and forecast_period coordinate.
+        """
+        result = self.plugin.process(self.cube, self.timestep)
+        self.assertEqual(result.coord("forecast_period").points, 600)
+        # 2017-11-10 04:10:00
+        self.assertEqual(result.coord("time").points, 1510287000)
+        self.assertEqual(
+            result.coord("forecast_reference_time").points, 1510286400)
+        self.assertEqual(result.coord("forecast_period").units, "seconds")
+        self.assertEqual(result.coord("time").units,
+                         "seconds since 1970-01-01 00:00:00")
+        self.assertEqual(result.coord("forecast_reference_time").units,
+                         "seconds since 1970-01-01 00:00:00")
+        self.assertEqual(result.coord("forecast_period").dtype, np.int32)
+        self.assertEqual(result.coord("time").dtype, np.int64)
+        self.assertEqual(
+            result.coord("forecast_reference_time").dtype, np.int64)
+
+    def test_time_unit_conversion(self):
+        """Test that the output cube contains the desired units and datatypes
+        for the time, forecast_reference_time and forecast_period coordinate,
+        where a unit conversion has been required."""
+        self.cube.coord("time").convert_units("hours since 1970-01-01 00:00")
```

Comment:
Could you also please have this cover `forecast_reference_time` in hours.

Reply:
I've amended this test to cover `forecast_reference_time` with units of "hours since 1970".